notes_app/client/app/notes/notes_list.component.js

```diff
@@ -15,11 +16,11 @@
         notesService.pagenumToOffset(ctrl.currentPage, ctrl.noteLimit)
       ).then(function(response) {
         ctrl.metadata = response.meta;
-        ctrl.notes = response.objects;
+        ctrl.notes = notesService.data.notes;
```

Comment:
I think you can move this line above, out of `list`'s `then`. And you can keep `metadata` in service's `data` and map controller's `metadata` to it.